src/components/NavBar/LoginSignIn/SignInComponent/SignIn.component.jsx

```diff
@@ -0,0 +1,151 @@
+
+  const handlePassword = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const tmpPassword = e.target.value;
+    setPassword(tmpPassword);
+  };
+
+  const handleConfirmatoryPassword = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const tmpPassword = e.target.value;
+    setConfirmatoryPassword(tmpPassword);
+  };
+
+  const arePassWordsEqual = password.length > 6 && password === confirmatoryPassword;
+  const isEmailAndPasswordFilled = email.length > 0 && arePassWordsEqual;
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    let isMounted = true;
+
+    if (!arePassWordsEqual) {
+      setPassword('');
+      setConfirmatoryPassword('');
+      return;
+    }
+
+    if (!isEmailAndPasswordFilled) {
+      setErrorOnSubmit(true);
+      return;
+    }
+
+    try {
+      await auth.createUserWithEmailAndPassword(email, password);
+      if (isMounted) setRegisteredSuccessfully(true);
+    } catch (error) {
+      console.log(error);
+      if (isMounted) {
+        setErrorMessage(error.message);
+        setEmail('');
+        setPassword('');
+        setConfirmatoryPassword('');
+        setErrorOnSubmit(true);
+      }
+    }
+
+    return () => {
+      isMounted = false;
+    };
+  };
+
+  return (
+    <div>
+      <Styled.LoginContainer>
+        <Styled.StyledForm onSubmit={handleSubmit}>
+          {errorOnSubmit && (
+            <Styled.AlertFlag>{errorMessage || 'Something went wrong'}</Styled.AlertFlag>
+          )}
+
+          <Styled.InputField>
+            <Styled.Label htmlFor="user-name">
+              Email<Styled.RequiredField>*</Styled.RequiredField>:
+            </Styled.Label>
+            <Styled.UserEmailInput
+              id="user-name"
+              onChange={handleUserEmail}
+              value={email}
+            />
+          </Styled.InputField>
+
+          <Styled.InputField>
+            <Styled.Label htmlFor="user-password">
+              Password<Styled.RequiredField>*</Styled.RequiredField>:
+            </Styled.Label>
+            <Styled.UserPasswordInput
+              id="user-password"
+              onChange={handlePassword}
+              value={password}
+            />
+          </Styled.InputField>
+
+          <Styled.InputField>
+            <Styled.Label htmlFor="user-confirmatory-password">
+              Password<Styled.RequiredField>*</Styled.RequiredField>:
+            </Styled.Label>
+            <Styled.UserPasswordInput
+              id="user-confirmatory-password"
+              onChange={handleConfirmatoryPassword}
+              value={confirmatoryPassword}
+            />
+          </Styled.InputField>
+
+          {password.length <= 6 && (
+            <Styled.AlertFlag>Password must have at least 7 characters</Styled.AlertFlag>
+          )}
+
+          {!arePassWordsEqual &&
+            password.length > 6 &&
+            confirmatoryPassword.length > 6 && (
```

Comment:
These validations were implied in `arePasswordsEqual`. There, you're already checking for the password length. Also, I don't think you should be checking length there, since you have a message specific for that error in line 131. The only concern for this message should be if the passwords are equal or not, regardless the length.